it/src/main/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManager.java

```diff
@@ -0,0 +1,390 @@
+    try (BigtableDataClient dataClient =
+        bigtableResourceManagerClientHandler.getBigtableDataClient()) {
+      for (RowMutation tableRow : tableRows) {
+        dataClient.mutateRow(tableRow);
+      }
+    } catch (Exception e) {
+      throw new BigtableResourceManagerException("Failed to write mutations.", e);
+    }
+
+    LOG.info("Successfully sent mutations to instance {}.", instanceId);
+  }
+
+  @Override
+  public synchronized ImmutableList<Row> readTable(String tableId) {
+    checkHasInstance();
+    checkHasTable(tableId);
+
+    // List to store fetched rows
+    ImmutableList.Builder<Row> tableRowsBuilder = ImmutableList.builder();
+
+    LOG.info("Reading all rows from {}.{}", instanceId, tableId);
+
+    // Fetch the Bigtable data client and read all the rows from the table given by tableId
+    try (BigtableDataClient dataClient =
+        bigtableResourceManagerClientHandler.getBigtableDataClient()) {
+
+      Query query = Query.create(tableId);
+      ServerStream<Row> rowStream = dataClient.readRows(query);
+      for (Row row : rowStream) {
+        tableRowsBuilder.add(row);
+      }
+
+    } catch (IOException | NotFoundException e) {
+      throw new BigtableResourceManagerException("Error occurred while reading table rows.", e);
+    }
+
+    ImmutableList<Row> tableRows = tableRowsBuilder.build();
+    LOG.info("Loaded {} rows from {}.{}", tableRows.size(), instanceId, tableId);
+
+    return tableRows;
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    LOG.info("Attempting to cleanup manager.");
+    try {
+      bigtableResourceManagerClientHandler
+          .getBigtableInstanceAdminClient()
+          .deleteInstance(instanceId);
+      bigtableResourceManagerClientHandler.getBigtableInstanceAdminClient().close();
+      bigtableResourceManagerClientHandler.getBigtableTableAdminClient().close();
+      hasInstance = false;
+    } catch (Exception e) {
+      throw new BigtableResourceManagerException("Failed to delete resources.", e);
+    }
+
+    LOG.info("Manager successfully cleaned up.");
+  }
+
+  /** Builder for {@link DefaultBigtableResourceManager}. */
+  public static final class Builder {
+
+    private String testId;
+    private String projectId;
+    private String instanceId;
+    private CredentialsProvider credentialsProvider;
+
+    private BigtableResourceManagerClientHandler bigtableResourceManagerClientHandler;
+
+    private Builder(String testId, String projectId) {
+
+      this.testId = testId;
+      this.projectId = projectId;
+      this.instanceId = generateInstanceId(testId);
+    }
+
+    public Builder setTestId(String testId) {
+      this.testId = testId;
+      return this;
+    }
+
+    public String getTestId() {
+      return testId;
+    }
+
+    public Builder setProjectId(String projectId) {
+      this.projectId = projectId;
+      return this;
+    }
+
+    public String getProjectId() {
```

Comment:
Builders are typically made write-only. I'd remove all getters.
Also consider using `@AutoBuilder`, which generates a lot of boilerplate code for you.

Reply:
Fixed.